File: admin/src/type/category.ts
// src/type/category.ts

export interface Category {
  id: number;
  parentId?: number | null;
  name: string;
  slug: string;
  imageUrl?: string | null;
  sortOrder: number;
  description?: string | null;
  createdAt: string;
}

export interface CategoryPage {
  content: Category[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number; // page index
}

export interface ApiResponse<T> {
  success: boolean;
  message?: string;
  data: T;
}

File: BE/fashion-api/src/main/java/com/tlcn/fashion_api/entity/product/Category.java
package com.tlcn.fashion_api.entity.product;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "categories")
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "parent_id", insertable = false, updatable = false)
    private Long parentId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    @JsonIgnore
    private Category parent;

    private String name;
    private String slug;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "sort_order")
    private Integer sortOrder;

    @Column(name = "created_at", nullable = false, updatable = false)
    @CreationTimestamp
    private LocalDateTime createdAt;

    private String description;

    @OneToMany(mappedBy = "category")
    @JsonIgnore
    private java.util.Set<ProductCategory> productCategories;
}
